src/main/scala-2.13+/com/risksense/ipaddr/IpSet.scala

```diff
@@ -0,0 +1,482 @@
+  def >=(that: IpSet): Boolean = this.supersetOf(that) // scalastyle:ignore method.name
+
+  /** Checks if this IpSet is superset of another IpSet.
+    *
+    * @param that an IpSet
+    * @return True if `this` is superset of `that`, False otherwise.
+    */
+  def supersetOf(that: IpSet): Boolean = that.subsetOf(this)
+
+  /** Checks if this IpSet is subset of another IpSet.
+    *
+    * @param that an IpSet
+    * @return True if `this` is subset of `that`, False otherwise.
+    */
+  def subsetOf(that: IpSet): Boolean = this.networkSeq.forall(that.contains(_))
+
+  /** Test if a given [[IpAddress]] is present in this IpSet.
+    *
+    * @param ipAddress an IpAddress object
+    * @return True if the input IpAddress is present in this IpSet, False otherwise.
+    */
+  def contains(ipAddress: IpAddress): Boolean = contains(IpNetwork(ipAddress, ipAddress.width))
+
+  /** Test if a given [[IpNetwork]] is present in this IpSet.
+    *
+    * @param network a IpNetwork object
+    * @return True if input IpNetwork is present in this IpSet, False otherwise.
+    */
+  def contains(network: IpNetwork): Boolean = this.exists(_.contains(network))
+
+  /** Checks if this IpSet has nothing in common with another IpSet.
+    *
+    * @param that an IpSet
+    * @return True if `this` and `that` have no common elements, False otherwise.
+    */
+  def isDisjoint(that: IpSet): Boolean = this.intersect(that).isEmpty
+
+  /** Symmetric difference
+    *
+    * Returns an IpSet containing networks that appear in either `this` or `that` but not in both.
+    *
+    * @param that an IpSet
+    * @return an IpSet resulting from symmetric difference
+    */
+  def ^(that: IpSet): IpSet = this.symmetricDiff(that) // scalastyle:ignore method.name
+
+  /** Symmetric difference
+    *
+    * Returns an IpSet containing networks that appear in either `this` or `that` but not in both.
+    *
+    * @param that an IpSet
+    * @return an IpSet resulting from symmetric difference
+    */
+  def symmetricDiff(that: IpSet): IpSet = {
+    val common = this.intersect(that)
+    val all = this | that
+    val res = all.diff(common)
+    IpSet(res.toSeq)
+  }
+
+  /** Intersection of `this` IpSet and `that`
+    *
+    * @param that an IpSet to perform intersection with
+    * @return an IpSet containing all networks that are common between `this` IpSet and `that`.
+    */
+  def intersect(that: IpSet): IpSet = {
+    val thisNets = this.networkSeq
+    val thatNets = that.networkSeq
+
+    /* Recursively find common networks
+     *
+     * @param s1 `this` IpSets network sequence
+     * @param s2 `that` IpSets network sequence
+     * @param result a sequence of IpNetwork objects common in `this` and `that`
+     */
+    @tailrec
+    def intersectRecurse(
+        s1: Seq[IpNetwork],
+        s2: Seq[IpNetwork],
+        result: Seq[IpNetwork]): Seq[IpNetwork] = {
+      if (s1.isEmpty || s2.isEmpty) {
+        result
+      } else {
+        if (s1.head == s2.head) {
```

Comment:
Anyway to remove calls to `.head` ?
```suggestion
        if (s1.headOption == s2.headOption) {
```
Should also work no?

Reply:
It is not the goal of this PR to change behaviour like this.

Reply:
OK, so I take it that this was already coded this way (didn't check 😶 ) 👍